src/Controller/ExportController.php

```diff
@@ -4,69 +4,59 @@
+use FINDOLOGIC\FinSearch\Export\ProductIdExport;
+use FINDOLOGIC\FinSearch\Export\ProductService;
+use FINDOLOGIC\FinSearch\Export\SalesChannelService;
+use FINDOLOGIC\FinSearch\Export\XmlExport;
+use FINDOLOGIC\FinSearch\Logger\Handler\ProductErrorHandler;
+use FINDOLOGIC\FinSearch\Struct\Config;
+use FINDOLOGIC\FinSearch\Validators\ExportConfiguration;
 use Psr\Log\LoggerInterface;
-use Shopware\Core\Checkout\Customer\Aggregate\CustomerGroup\CustomerGroupEntity;
-use Shopware\Core\Content\Product\Aggregate\ProductVisibility\ProductVisibilityDefinition;
-use Shopware\Core\Content\Product\ProductEntity;
-use Shopware\Core\Content\Product\SalesChannel\ProductAvailableFilter;
-use Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException;
-use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
-use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
-use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
-use Shopware\Core\Framework\DataAbstractionLayer\Search\IdSearchResult;
 use Shopware\Core\Framework\Routing\Annotation\RouteScope;
 use Shopware\Core\System\SalesChannel\Context\SalesChannelContextFactory;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
-use Shopware\Core\System\SystemConfig\SystemConfigEntity;
-use Shopware\Core\System\SystemConfig\SystemConfigService;
 use Shopware\Storefront\Framework\Routing\Router;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 use Symfony\Component\Routing\RouterInterface;
-use Symfony\Component\Validator\Constraints as Assert;
-use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\ConstraintViolation;
 use Symfony\Component\Validator\Validation;
-use Throwable;
 
-class ExportController extends AbstractController implements EventSubscriberInterface
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExportController extends AbstractController
 {
-    private const DEFAULT_START_PARAM = 0;
-    private const DEFAULT_COUNT_PARAM = 20;
-
     /** @var LoggerInterface */
     protected $logger;
 
     /** @var Router */
     private $router;
 
-    /**
-     * @var HeaderHandler
-     */
+    /** @var HeaderHandler */
     private $headerHandler;
 
-    /**
-     * @var SalesChannelContext
-     */
+    /** @var SalesChannelContextFactory */
+    private $salesChannelContextFactory;
+
+    /** @var SalesChannelContext */
     private $salesChannelContext;
 
-    private $salesChannelContextFactory;
+    /** @var ExportConfiguration */
+    private $config;
+
+    /** @var ProductService */
+    private $productService;
+
+    /** @var Config */
+    private $pluginConfig;
+
+    /** @var Export|XmlExport|ProductIdExport */
+    private $export;
```

Comment:
Is this comment supposed to be here? I do not get what you mean with this.

Reply:
As the classes are also called `*Export` we can leave it as export. This also relates to this [comment](https://github.com/findologic/plugin-shopware-6/pull/94#discussion_r517422608). We do have export, exporter, maybe more similar wordings. If you are not deep in the code it's hard to tell what those classes do, calling export url, fetching products or generating xml?!

Reply:
> If you are not deep in the code it's hard to tell what those classes do, calling export url, fetching products or generating xml?!

No class that has XML in the name is fetching any products. In this case I went for a generic `export`, since the exporter implementations either return an `XML` or a `JSON`, in case a product is invalid, or some fields are not valid. The only class, where it really may get confusing is the `XmlExport`, since it has `XMLExporter` from libflexport, as an optional argument. Maybe I even drop that, as it isn't really used anywhere anyway.

Reply:
I will leave this parameter optional. This way you'd have the possibility to use your own libflexport XMLExporter.

Reply:
tbd

Reply:
Rename XmlExporter as discussed